### ISSUE TITLE
samtools cat, merge headers

### ISSUE THREAD

**Issue by dozy** [samtools/samtools#959]
It would be useful if the samtools cat command merged headers (in the way that samtools merge does)

**Comment by finswimmer** [comment]
Yes, indeed this would be useful. Or to say it in other words: It would useful if `samtools`would offer a way to merge/concatenate unsorted files including the headers.

**Comment by daviesrob** [comment]
Are there any particular headers that you would like to merge?  `samtools cat` would get a lot slower if you want to do a full merge, but it might be possible to preserve its speed in simpler cases.

In the most complicated case, it would pretty much do the same as `samtools merge`, but adding the files one after the other.

**Comment by feihongloveworld** [comment]
yes，it‘s very usefull ，because bams have diff @RG，

**Comment by Krischan** [comment]
Second this. It would be quite useful to rapidly concatenate multiple unmapped bam files for the same sample, originating from multiple Illumina flow cells and/or lanes. The resulting bam would still be unmapped, but carry along the read group in the header (RG) and all other tags.
Minimally, the header RG tag for each BAM file should be propagated.
At the moment I am doing this with `Picard MergeSamFiles ASSUME_SORTED=TRUE`, which is much slower.
